### PR TITLE
Set Service Category data

### PR DESCRIPTION
**Pull request by karenmillermoj**
## What does this pull request do?

Sets up a service_category table with the initial Dynamic Framework Intervention Service Categories and provides the service and repository access ready for controller and api integration

## What is the intent behind these changes?

Allow the initial service categories required to be set up in the database

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/service/ServiceCategoryService.kt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import java.util.UUID
+
+@Service
+class ServiceCategoryService(val repository: ServiceCategoryRepository) {
+
+  fun getServiceCategory(id: UUID): ServiceCategory? {
+    return repository.findByIdOrNull(id)
+  }
+
+  fun getServiceCategories(): MutableIterable<ServiceCategory> {
```
Comment (tomsmyers) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/service/ServiceCategoryService.kt:16]: why is this mutableiterable?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/service/ServiceCategoryService.kt [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.ServiceCategory
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository.ServiceCategoryRepository
+import java.util.UUID
+
+@Service
+class ServiceCategoryService(val repository: ServiceCategoryRepository) {
+
+  fun getServiceCategory(id: UUID): ServiceCategory? {
```
Comment (tomsmyers) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/service/ServiceCategoryService.kt:12]: nitpick: prefer `getServiceCategoryByID` or similar

File: src/main/resources/db/migration/V1_3__seed_service_category.sql [hunk 3]
```diff
@@ -0,0 +1,9 @@
+insert into service_category (id, name) values ('d21fc249-41ff-4987-8027-0c559f9bbd2e', 'Accommodation');
```
Comment (tomsmyers) [inline src/main/resources/db/migration/V1_3__seed_service_category.sql:1]: the uuid for this category currently in use in the pact tests is `428ee70f-3001-4399-95a6-ad25eaaede16`, easiest to match it here.

File: src/main/resources/db/migration/V1_3__seed_service_category.sql [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+insert into service_category (id, name) values ('ca374ac3-84eb-4b91-bea7-9005398f426f','Education, Training and Employment (ETE)');
```
Comment (tomsmyers) [inline src/main/resources/db/migration/V1_3__seed_service_category.sql:2]: does created timestamp get set when these are not created by hibernate?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/repository/ServiceCategoryRepositoryIntegrationTest.kt [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository.ServiceCategoryRepository
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ActiveProfiles("test", "local")
+class ServiceCategoryRepositoryIntegrationTest(@Autowired private val repository: ServiceCategoryRepository) {
+
+  @Test
+  fun `check entered`() {
+    val result = repository.findAll()
+    assertThat(result.count()).isEqualTo(9)
```
Comment (tomsmyers) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/repository/ServiceCategoryRepositoryIntegrationTest.kt:17]: this test will break if add more service categories. what are we actually trying to test?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/repository/ServiceCategoryRepositoryTest.kt [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+@DataJpaTest
+@ActiveProfiles("db")
+class ServiceCategoryRepositoryTest @Autowired constructor(
+  val entityManager: TestEntityManager,
+  val serviceCategoryRepository: ServiceCategoryRepository
+) {
+
+  @Test
+  fun `when findByIdOrNull then return referral`() {
```
Comment (tomsmyers) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/repository/ServiceCategoryRepositoryTest.kt:19]: this is almost identical to the 'service' test - is it necessary?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/repository/ServiceCategoryRepositoryIntegrationTest.kt (hunk 5 above)
Comment (karenmillermoj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/repository/ServiceCategoryRepositoryIntegrationTest.kt:17]: Was my initial test to ensure if it was all set ok.  I am not happy with with test anyway, so I will remove it.

File: src/main/resources/db/migration/V1_3__seed_service_category.sql (hunk 3 above)
Comment (karenmillermoj) [reply src/main/resources/db/migration/V1_3__seed_service_category.sql:1]: Definitely.  Updated

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/service/ServiceCategoryService.kt (hunk 2 above)
Comment (karenmillermoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/service/ServiceCategoryService.kt:12]: No worries - happy to change

File: src/main/resources/db/migration/V1_3__seed_service_category.sql (hunk 4 above)
Comment (karenmillermoj) [reply src/main/resources/db/migration/V1_3__seed_service_category.sql:2]: No.  Good spot - I will add that in

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/service/ServiceCategoryService.kt (hunk 1 above)
Comment (karenmillermoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/service/ServiceCategoryService.kt:16]: Have swapped to return a List<ServiceCategory> - using toList()

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/repository/ServiceCategoryRepositoryTest.kt (hunk 6 above)
Comment (karenmillermoj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/repository/ServiceCategoryRepositoryTest.kt:19]: You're right. Removed

**Review by tomsmyers — APPROVED** [review]
👍

**Review by tomsmyers — APPROVED** [review]

**Pull request merged**